### ISSUE TITLE
additional sampling scheme

### ISSUE THREAD

**Issue by vseledkin** [ChunyuanLI/Optimus#6]
Currently, text is generated from latent point by sampling from distributions produced by generator over vocabulary of tokens. But since z is multivariate gaussian we can also sample from it thus having more diversity and nuances in generated samples.

**Comment by ChunyuanLI** [comment]
You are right about my current implementation, and I agree it will increase diversity. I‘m bit afraid that adding randomness in latent space may delude the latent trajectory, one has to carefully choose random samples by checking its distance to its Gaussian mean.
